packages/ui-components/src/Input/input.scss

```diff
@@ -0,0 +1,92 @@
+@import "../styles/tokens.scss";
+@import "./constants.scss";
+
+// ".container" class is responsible to style up outer borders
+// and width, height sizes of input component.
+// it can be applied to <input/> element itself or to wrapper <div>
+// element in case NumberInput is rendered.
+.container {
+  border-radius: $input-border-spacing;
+  height: exclBorderWidth($input-height);
+  display: inline-flex;
+  background-color: $crl-base-white;
+  min-width: exclBorderWidth($input-text-width);
+  width: exclBorderWidth($input-text-width);
+  padding: $input-border-spacing;
+
+  &.active {
+    border: solid $border-width $crl-neutral-4;
+    background-color: $crl-base-white;
+    color: initial;
+
+    &:hover {
+      border: solid $border-width $crl-neutral-5;
+    }
+
+    &:focus-within {
+      border: solid $border-width $crl-base-blue;
+      // Outline params are borrowed from default outline styles for input element
+      // and applied to div element.
+      outline: $crl-base-blue auto 3px;
+      outline-offset: -2px;
+    }
+  }
+
+  &.invalid {
+    border: solid $border-width $crl-base-red;
+    background-color: $crl-red-1;
+    color: $crl-base-red;
+
+    &:focus-within, &:hover {
+      border: solid $border-width $crl-base-red;
+    }
+  }
+
+  &.disabled {
+    background: $crl-neutral-1;
+    color: $crl-base-text--light;
+    border: solid $border-width $crl-neutral-3;
+    cursor: default;
+
+    &:focus, &:hover {
+      border: solid $border-width $crl-neutral-3;
+    }
+  }
+}
+
+// ".input" removes most styling from element because it is responsibility of `.container` class.
+.input {
+  font-family: Source Sans Pro, sans-serif; // TODO (koorosh): replace with token value
+  font-size: 14px; // TODO (koorosh): replace with token value (default font size)
+  line-height: 22px;
+  border: none;
+  outline: none;
+  padding: 0 crl-gutters(1.5);
+  position: relative;
+  background-color: inherit;
+  color: inherit;
+  width: 100%;
+
+  &:focus, &:hover {
+    border: none;
+    outline: none;
+  }
+}
+
+// Prefix component styles. Set constant width/height
+// and apply colors based on container's state.
+.container {
+  .prefix {
+    width: 14px; // TODO (koorosh): has to be the same as font-size
+    min-width: 14px;
+    height: 14px;
```

Comment:
In Firefox, the prefix icon is misaligned

<img width="304" alt="Screen Shot 2020-05-21 at 12 38 22" src="https://user-images.githubusercontent.com/397448/82582425-08801980-9b60-11ea-8ca9-f758561780f7.png">

```suggestion
    height: 22px;
    line-height: 22px;
```

Reply:
@nathanstilwell , I couldn't reproduce this issue. 
for this element, `margin-top` and `margin-bottom` are set to auto - so it has to do a trick and center element.
`height` and `width` define the size of SVG icon and it has to be the same as font size.
Please, let me know if there are some specific conditions required to reproduce this.

Reply:
Interesting. I’m using FF developer edition, but if the issue isn’t present in the stable version of FF, I wouldn’t worry about it.

I do have a concern that all of the icons are not square and so wonder if that will effect how they are rendered. This is also an issue I would like to address from the Icon side as having square icons that are all the same base size will make this kind of thing simpler. @Annebirzin

Reply:
@nathanstilwell yep agree with making all icons square. I can work with you on this